skrub/_gap_encoder.py

```diff
@@ -430,8 +435,26 @@ def _check_analyzer(self):
             raise ValueError("analyzer should be one of ['word', 'char', 'char_wb'].")
 
     def fit(self, X, y=None):
+        """Fit the GapEncoder and transform a column.
+
+        Parameters
+        ----------
+        X : Column, shape (n_samples, )
+            The string data to fit the model on.
+        y : None
+            Unused, only here for compatibility.
+
+        Returns
+        -------
+        GapEncoder
+            The fitted GapEncoder instance (self).
         """
-        Fit the GapEncoder on `X`.
+        _ = self.fit_transform(X)
```

Comment:
I worry that calling fit_transform when we only need fit leads to additional costs, as it is possible to do "fit" without the transform step